### ISSUE TITLE
[pickers] Date pickers custom columns example blocks picking day from keyboard focus

### ISSUE THREAD

**Issue by StylesTrip** [mui/mui-x#17506]
### Steps to reproduce

Steps:
1. Open this link to live example: https://mui.com/x/react-data-grid/custom-columns/#date-pickers
2. Open the filter panel in the date pickers data grid example
3. With the keyboard, assuming the column that is selected is "Last Login" then keyboard tab over to the date time picker icon to open the date time picker
4. Observe that the keyboard focus is on the time picker section. Try to keyboard tab to the day picker.
5. Or try to select a month/year and see where the focus goes

### Current behavior

1. Day picker is not keyboard accessible within the filter panel's date time picker
2. Year picker is accessible but upon selected the year picker the focus is brought back to the time picker section

https://github.com/user-attachments/assets/289bb9b7-a2b8-4aeb-8a6d-d5833763d643

### Expected behavior

Within the Data Grid filter panel, the date time picker's day picker should be in the tab order and be accessible from the keyboard to select a day.
The same goes for year/month picker.

### Context

Keyboard accessibility in the date time picker in the filter panel for data grids.

One potential solution may be related to removing the autoFocus from this section:

```jsx
function GridFilterDateInput(props) {
  const { item, showTime, applyValue, apiRef } = props;

  const Component = showTime ? DateTimePicker : DatePicker;

  const handleFilterChange = (newValue) => {
    applyValue({ ...item, value: newValue });
  };

  return (
    <Component
      value={item.value ? new Date(item.value) : null}
      autoFocus
      label={apiRef.current.getLocaleText('filterPanelInputLabel')}
      slotProps={{ textField: { size: 'small' } }}
      onChange={handleFilterChange}
    />
  );
}
```

I have a CodeSandbox example that removes the autoFocus entirely from the example that seems to have resolved the issue: https://codesandbox.io/p/sandbox/wandering-morning-mcdqf6?file=%2Fsrc%2FDemo.js

Not sure if there is a more elegant way to fix this.

### Your environment

  System:
    OS: macOS 14.5
  Browser:
    Chrome: 135.0.7049.96

**Search keywords**: keyboard focus data grid datepickers

**Comment by michelengelen** [comment]
Hey @StylesTrip and thanks for opening the issue here.
I can see that your suggested fix solves the problem of the filter input. Nice job on that!
I did encounter another problem with the edit input though.
Hitting tab will move the focus to the next cell, so we would need to intercept that behavior somehow to make this example fully accessible with keyboards.
Again: Thanks for raising this topic!

**Comment by LukasTy** [comment]
@StylesTrip did the `DateTimePicker` work differently before updating to v8?
This looks like an issue on Pickers and one that existed well before (even v6 behaves like this). 🙈

**Comment by LukasTy** [comment]
Identified this as a regression + bug on Pickers v8.
We have two problems here:
1. A regression on view switching without `autoFocus` - focused view remains on `DateCalendar` after selection, whereas it should move to `MultiSectionDigitalClock` as it does on v7.
2. The `autoFocus` prop is affecting the `focusedView` behavior, it forces the focus to remain on the `MultiSectionDigitalClock`.
